tests/components/zha/test_light.py

```diff
@@ -63,6 +67,44 @@
 }
 
 
+@asynctest.mock.patch(
+    "zigpy.zcl.clusters.general.OnOff.read_attributes", new=MagicMock()
+)
+async def test_light_refresh(hass, zigpy_device_mock, zha_device_joined_restored):
+    """Test zha light platform refresh."""
+
+    # create zigpy devices
+    zigpy_device = zigpy_device_mock(LIGHT_ON_OFF)
+    zha_device = await zha_device_joined_restored(zigpy_device)
+    on_off_cluster = zigpy_device.endpoints[1].on_off
+    entity_id = await find_entity_id(DOMAIN, zha_device, hass)
+
+    # allow traffic to flow through the gateway and device
+    await async_enable_traffic(hass, [zha_device])
+    on_off_cluster.read_attributes.reset_mock()
+
+    # not enough time passed
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(minutes=20))
+    await hass.async_block_till_done()
+    assert on_off_cluster.read_attributes.call_count == 0
+    assert on_off_cluster.read_attributes.await_count == 0
+    assert hass.states.get(entity_id).state == STATE_OFF
+
+    # 1 interval - 1 call
+    async_fire_time_changed(hass, dt_util.utcnow() + timedelta(minutes=80))
+    await hass.async_block_till_done()
+    assert on_off_cluster.read_attributes.call_count == 1
+    assert on_off_cluster.read_attributes.await_count == 1
+    assert hass.states.get(entity_id).state == STATE_OFF
```

Comment:
We should have the state update include some specific state attributes that are different between each interval so we can tell that the interval has actually changed something.

Reply:
In reality there is no guarantee that there are any changes... and in most cases there probably won't be. This is to catch cases where someone hard powered off a bulb or controlled it with a directly bound remote and changed the dimming level or the color etc. for bulbs such as hue that don't report state as the spec says they should. Honestly... we have already proven the code that was added works and this is all waste in a test that just adds more test code with no benefit. I'll make the change as requested but this really seems fruitless IMO. Is there anything else you think I should add to this?

Reply:
We need to assert that the entity updates state correctly on the interval. If we would have added this from the beginning we would have seen that there was a missing state update in the refresh method.

I don't think this is a waste of time.

Reply:
done